src/verinfast/utils/utils.py

```diff
@@ -147,3 +147,36 @@ def log(self, msg, tag=None, display=False, timestamp=True):
             f.write(output+"\n")
         if display or self.debug:
             print(output)
+
+
+# Returns a tuple of the repo name, repo url and branch from the original url
+def get_repo_name_url_and_branch(repo_url: str):
+    # The regex pattern captures the entire URL
+    # and the repository name after the last '/'
+
+    repo_url = repo_url.strip()
+    branch = None
+
+    split_url = repo_url.split("@")
+
+    # HTTPS URLs
+    if repo_url.startswith("https://") or repo_url.startswith("http://"):
```

Comment:
basic auth can be used with html, e.g.

https://user:password@my.githost.com/repo@branch

Reply:
added to unit test and then added code to handle.